mirrord-layer/src/file/hooks.rs

```diff
@@ -34,14 +35,16 @@ pub(super) unsafe extern "C" fn open_detour(raw_path: *const c_char, open_flags:
         libc::open(raw_path, open_flags)
     } else {
         let open_options = OpenOptionsInternalExt::from_flags(open_flags);
-        let open_result = open(path, open_options);
+        let open_result = open(path.clone(), open_options);
 
-        open_result
-            .map_err(|fail| {
-                error!("Failed opening file with {:#?}", fail);
+        match open_result {
+            Ok(fd) => fd,
+            Err(fail) => {
+                error!("Failed opening file {:#?} with {:#?}", path, fail);
+                set_errno(Errno(fail.into()));
```

Comment:
Have you tried using `set_errno` in the `From` impl instead of in each function?